pkg/operator/ceph/object/objectstore.go

```diff
@@ -205,14 +205,6 @@ func deleteSingleSiteRealmAndPools(objContext *Context, spec cephv1.ObjectStoreS
 // This is used for quickly getting the name of the realm, zone group, and zone for an object-store to pass into a Context
 func getMultisiteForObjectStore(ctx context.Context, clusterdContext *clusterd.Context, spec *cephv1.ObjectStoreSpec, namespace, name string) (string, string, string, error) {
 
-	if spec.IsExternal() {
-		// Currently external cluster with zones/zonegroup/realm are not supported, will be
-		// fixed by https://github.com/rook/rook/issues/6342. So if user does not create
-		// zone/zonegroup, RGW internally creates zone/zonegroup named as `default`, hence
-		// Rook can set this value in the object context fields.
-		return "", "default", "default", nil
-	}
-
 	if spec.IsMultisite() {
```

Comment:
@travisn Is this checks the particular CR name https://github.com/rook/rook/blob/master/deploy/examples/object-multisite.yaml? https://github.com/rook/rook/blob/2e89649b3a64b17337203e73a1852ba4bfe15bc4/pkg/apis/ceph.rook.io/v1/object.go#L36
If not, if the CR is not created then it will always fail as won't be able to get as a resource.

I will also take a look once more.